packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts

```diff
@@ -229,12 +235,14 @@ export class SharedTreeCore<TEditor extends ChangeFamilyEditor, TChange> extends
 	/**
 	 * Submits an op to the Fluid runtime containing the given commit
 	 * @param commit - the commit to submit
+	 * @returns the submitted commit. This is undefined if the underlying `SharedObject` is not attached,
+	 * and may differ from `commit` due to enrichments like detached tree refreshers.
 	 */
-	private submitCommit(
+	protected submitCommit(
 		commit: GraphCommit<TChange>,
 		schemaAndPolicy: ClonableSchemaAndPolicy,
 		isResubmit = false,
-	): void {
+	): GraphCommit<TChange> | undefined {
```

Comment:
Alright, since we're having other fascinating language etiquette discussions on this PR, let's have another one here :D

What are your thoughts on "implicitly returning undefined" (as this function sometimes does) as opposed to always writing `return undefined;` where applicable? I am undecided and I am curious for your thoughts. Historically it seemed to me that we would always lean towards explicit returns (didn't we used to have a lint rule for that or something?), but I don't know whether that was a good thing or not.

Reply:
I do prefer the explicit return, but I really don't mind too much.